src/librustc_target/abi/mod.rs

```diff
@@ -619,13 +619,16 @@ impl Scalar {
 /// Describes how the fields of a type are located in memory.
 #[derive(PartialEq, Eq, Hash, Debug, HashStable_Generic)]
 pub enum FieldsShape {
+    Primitive,
+
     /// All fields start at no offset. The `usize` is the field count.
-    ///
-    /// In the case of primitives the number of fields is `0`.
     Union(usize),
```

Comment:
Now the number of fields is never 0, right? Are there good places to assert this? Or maybe use `NonZeroUsize` to encode this on the type level?

Reply:
`NonZeroUsize` would be great, if nothing else it would prevent misuse in the future.